pkg/retry/retry.go

```diff
@@ -0,0 +1,69 @@
+package retry
+
+import (
+	"log"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/linode/linodego/pkg/errors"
+)
+
+const retryAfterHeaderName = "Retry-After"
+
+// ConditionFunc represents a function that evaluates whether a request
+// should be retried, given the response and error.
+type ConditionFunc resty.RetryConditionFunc
+
+// ComposeConditional composes a ConditionFunc function from an array
+// of ConditionFunc functions.
+func ComposeConditionFunc(conditionals ...ConditionFunc) ConditionFunc {
```

Comment:
Did you intend to use this instead of the retry conditional above? Your intent I assume is that you could technically use a single retry condition without composing?

Reply:
Should also remove this, my initial intent was to use this in `(client).configureRetries` to step through `c.retryConditionals`, but I ended up not doing that so that I could keep our log before retrying.